hazelcast/src/main/java/com/hazelcast/map/impl/recordstore/MetadataStore.java

```diff
@@ -17,32 +17,83 @@
 package com.hazelcast.map.impl.recordstore;
 
 import com.hazelcast.internal.serialization.Data;
+import com.hazelcast.query.impl.AbstractMetadata;
 import com.hazelcast.query.impl.Metadata;
 
 import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.ConcurrentMap;
 
-public class MetadataStore {
+/**
+ * On-heap Metadata Store. Uses CHM to store the key/metadata pairs.
+ */
+public class MetadataStore implements AbstractMetadataStore {
```

Comment:
```suggestion
public class JsonMetadataStoreImpl implements JsonMetadataStore {
```

Reply:
Addressed in hazelcast/hazelcast-enterprise@1ca96b2 and b573cf6